Vowpal Wabbit - Contextual Bandits.ipynb

```diff
@@ -0,0 +1,705 @@
+      "To adjust logging level use sc.setLogLevel(newLevel). For SparkR, use setLogLevel(newLevel).\n",
+      "23/01/16 13:32:56 WARN SparkConf: The configuration key 'spark.yarn.user.classpath.first' has been deprecated as of Spark 1.3 and may be removed in the future. Please use spark.{driver,executor}.userClassPathFirst instead.\n",
+      "23/01/16 13:32:56 WARN SparkConf: The configuration key 'spark.yarn.user.classpath.first' has been deprecated as of Spark 1.3 and may be removed in the future. Please use spark.{driver,executor}.userClassPathFirst instead.\n",
+      "23/01/16 13:32:56 WARN SparkConf: The configuration key 'spark.yarn.user.classpath.first' has been deprecated as of Spark 1.3 and may be removed in the future. Please use spark.{driver,executor}.userClassPathFirst instead.\n",
+      "23/01/16 13:32:57 WARN Utils: Service 'SparkUI' could not bind on port 4040. Attempting port 4041.\n"
+     ]
+    }
+   ],
+   "source": [
+    "from pyspark.sql import SparkSession\n",
+    "\n",
+    "# Bootstrap Spark Session\n",
+    "spark = (\n",
+    "    SparkSession.builder.config(\n",
+    "        \"spark.jars.packages\",\n",
+    "        \"org.apache.hadoop:hadoop-azure:3.3.1,com.microsoft.azure:azure-storage:8.6.6,com.microsoft.azure:synapseml_2.12:0.10.2-114-409e395c-SNAPSHOT\",\n",
+    "    )\n",
+    "    .config(\"spark.jars.repositories\", \"https://mmlspark.azureedge.net/maven\")\n",
+    "    .config(\n",
+    "        \"spark.jars.excludes\",\n",
+    "        \"org.scala-lang:scala-reflect,org.apache.spark:spark-tags_2.12,org.scalatest:scalatest_2.12,com.fasterxml.jackson.core:jackson-databind\",\n",
+    "    )\n",
+    "    .config(\"spark.yarn.user.classpath.first\", \"true\")\n",
+    "    .getOrCreate()\n",
+    ")\n",
+    "\n",
+    "from synapse.ml.core.platform import *\n",
+    "\n",
+    "from synapse.ml.core.platform import materializing_display as display"
+   ]
+  },
+  {
+   "cell_type": "code",
+   "execution_count": 3,
+   "metadata": {},
+   "outputs": [
+    {
+     "name": "stderr",
+     "output_type": "stream",
+     "text": [
+      "23/01/16 13:34:49 WARN MetricsConfig: Cannot locate configuration: tried hadoop-metrics2-azure-file-system.properties,hadoop-metrics2.properties\n",
+      "[Stage 0:>                                                          (0 + 1) / 1]\r"
+     ]
+    },
+    {
+     "name": "stdout",
+     "output_type": "stream",
+     "text": [
+      "records read: 3\n",
+      "Schema: \n",
+      "root\n",
+      " |-- input: string (nullable = true)\n",
+      "\n"
+     ]
+    },
+    {
+     "name": "stderr",
+     "output_type": "stream",
+     "text": [
+      "\r",
+      "                                                                                \r"
+     ]
+    }
+   ],
+   "source": [
+    "import pyspark.sql.types as T\n",
+    "from pyspark.sql import functions as F\n",
+    "\n",
+    "schema = T.StructType(\n",
```

Comment:
i believe spark will auto infer the schema? or are ytou trying to keep it as json

Reply:
it needs to be kept as string. the JSON parsing happens inside VW